sentry-android-core/src/main/java/io/sentry/android/core/FrameMetrics.java

```diff
@@ -0,0 +1,102 @@
+package io.sentry.android.core;
+
+import org.jetbrains.annotations.ApiStatus;
+
+@ApiStatus.Internal
+final class FrameMetrics {
+
+  private int normalFrameCount;
+  private int slowFrameCount;
+  private int frozenFrameCount;
+
+  private long slowFrameDuration;
+  private long frozenFrameDuration;
+
+  public FrameMetrics() {}
+
+  public FrameMetrics(
+      int normalFrameCount,
+      int slowFrameCount,
+      long slowFrameDuration,
+      int frozenFrameCount,
+      long frozenFrameDuration) {
+    this.normalFrameCount = normalFrameCount;
+
+    this.slowFrameCount = slowFrameCount;
+    this.slowFrameDuration = slowFrameDuration;
+
+    this.frozenFrameCount = frozenFrameCount;
+    this.frozenFrameDuration = frozenFrameDuration;
+  }
+
+  public void addSlowFrame(long duration) {
+    slowFrameDuration += duration;
+    slowFrameCount++;
+  }
+
+  public void addFrozenFrame(long duration) {
+    frozenFrameDuration += duration;
+    frozenFrameCount++;
+  }
+
+  public void addNormalFrame(long duration) {
+    normalFrameCount++;
+  }
+
+  public int getNormalFrameCount() {
+    return normalFrameCount;
+  }
+
+  public int getSlowFrameCount() {
+    return slowFrameCount;
+  }
+
+  public int getFrozenFrameCount() {
+    return frozenFrameCount;
+  }
+
+  public long getSlowFrameDuration() {
+    return slowFrameDuration;
+  }
+
+  public long getFrozenFrameDuration() {
+    return frozenFrameDuration;
+  }
+
+  public int getTotalFrameCount() {
+    return normalFrameCount + slowFrameCount + frozenFrameCount;
+  }
+
+  public void clear() {
+    normalFrameCount = 0;
+
+    slowFrameCount = 0;
+    slowFrameDuration = 0;
+
+    frozenFrameCount = 0;
+    frozenFrameDuration = 0;
+  }
+
+  public FrameMetrics duplicate() {
```

Comment:
nit: we usually call it `clone()` in other places

Reply:
Yeah, that's on purpose. I didn't want to interfere with Object.clone() / the clonable interface here
<img width="578" alt="image" src="https://github.com/getsentry/sentry-java/assets/1411808/4836b3c6-9ad2-4587-98e9-99fe3879dba3">